### ISSUE TITLE
Desafio de ordenação com Map

### ISSUE THREAD

**Issue by diego-bernardesb** [cami-la/collections-java-api-2023#13]
Oi tudo bem, realizando o desfio de ordenação do com Map me deparei com um erro na hora de executar o método de remover livro, essa foi a solução que consegui chegar, em vez de criar uma variável do obj Livro, troquei para uma String, e em vez de passar o .getValue() passo o .getKey(), pois a IDE acabou dando um alerta dizendo que não conseguia encontra a key para realizar a remoção

Codigo que não funcionou:

![image](https://github.com/cami-la/collections-java-api-2023/assets/72576349/0d993d3e-49bb-4e08-b2ca-996125009e5b)

Codigo funcionou e removeu : 

![image](https://github.com/cami-la/collections-java-api-2023/assets/72576349/c2a7eb02-2a31-439a-b854-4582addaaa05)

**Comment by cami-la** [comment]
Show!! Ficou ótima sua solução. Parabéns!

Acabei de atualizar o repositório e resolvi este método da seguinte forma:

```java
  public void removerLivro(String titulo) {
    List<String> chavesRemover = new ArrayList<>();
    for (Map.Entry<String, Livro> entry : livros.entrySet()) {
      if (entry.getValue().getTitulo().equalsIgnoreCase(titulo)) {
        chavesRemover.add(entry.getKey());
      }
    }
    for (String chave : chavesRemover) {
      livros.remove(chave);
    }
  }
```

Fiz dessa forma, porque talvez tenha livros com chave diferentes, porém o nome do livro pode ser igual.
Vê se você entendeu minha lógica.

Qualquer dúvida, fico à disposição! (:

**Comment by diego-bernardesb** [comment]
Sim, ficou bem tranquilo de entender acabou que foi realizada mas um for pra dar uma segurança de remover apenas o livro correto de acordo com a chave, ficou muito bom

**Comment by cami-la** [comment]
Com `Stream API `fica ainda mais simples. 

```java
  public void removerLivro(String titulo) {
    livros.values()
        .removeIf(livro -> livro.getTitulo().equalsIgnoreCase(titulo));
```

No próximo curso comigo aqui na DIO você vai aprender sobre **Stream** e como facilita na manipulaçao das **Collections**.
